server/static/upload.js

```diff
@@ -179,6 +179,99 @@ $(function () {
         $(this).closest('.modal').trigger('fieldValidation');
     }
 
+    function copyFromTableToModal(tableRow, modal) {
+        for (var fieldName in fieldProperties) {
+            if (fieldProperties.hasOwnProperty(fieldName)) {
+                var destField = $('#edit-file-' + fieldName);
+                var fieldType = fieldProperties[fieldName]['type'];
+                var defaultValue = fieldProperties[fieldName]['default'];
+                var sourceValue = tableRow.find('.file-' + fieldName).text() || defaultValue;
+                if (fieldType === 'freetext') {
+                    destField.val(sourceValue);
+                } else if (fieldType === 'typeahead') {
+                    destField.typeahead('val', sourceValue);
+                }
+            }
+        }
+    }
+
+    function copyFromModalToTable(modal, tableRow) {
+        for (var fieldName in fieldProperties) {
+            if (fieldProperties.hasOwnProperty(fieldName)) {
+                var fieldType = fieldProperties[fieldName]['type'];
+                var value = '';
+                var field = $('#edit-file-' + fieldName);
+                if (!field.prop('disabled')) {
+                    if (fieldType === 'freetext') {
+                        value = field.val();
+                    } else if (fieldType === 'typeahead') {
+                        value = field.typeahead('val');
+                    }
+                }
+                tableRow.find('.file-' + fieldName).text(value);
+            }
+        }
+    }
+
+    function showUploadStateOptions() {
+        $('.cancel-upload, .progress, .resume-upload').show();
+        $('.option, .start-upload').hide();
+        $('.add-sample, .file-name, .resume-upload, .sample-name, .sample-option').addClass('disabled');
+    }
+
+    function hideUploadStateOptions() {
+        $('.option').show();
+        $('.cancel-upload, .progress').hide();
+        $('.add-sample, .resume-upload, .sample-option').removeClass('disabled');
+    }
+
+    function clearTableOfCompleted () {
+        var samples = $('.sample-section:first').nextUntil();
+
+        samples.each(function (i, value) {
+            if (clearSampleOfFiles($(value).attr('name'))) {
+                value.remove()
+            }
+        });
+    }
+
+    function clearSampleOfFiles(sampleName) {
+        var sampleTbody =  $('.sample-section[name="' + sampleName + '"]');
+        var sampleRow = sampleTbody.find('.sample-header-row');
+        var fileRows = sampleRow.nextUntil();
+        var identifiers = $.map(r.files, function (resumableFile) { return resumableFile.uniqueIdentifier });
```

Comment:
Why not use `r.getFromUniqueIdentifier` to determine if a given fileRow is errored out or not?